### PR TITLE
Requirements 5, 6 and 7

### PR DESCRIPTION
**Pull request by s10-lee**
### Requirement 5

Handle an arbitrary number of names as input. When `name` represents more than two names, separate them with commas and close with an Oxford comma and "and". For example, when `name` is `["Amy", "Brian", "Charlotte"]`, then the method should return the string `"Hello, Amy, Brian, and Charlotte."`

### Requirement 6
Allow mixing of normal and shouted names by separating the response into two greetings. For example, when `name` is `["Amy", "BRIAN", "Charlotte"]`, then the method should return the string `"Hello, Amy and Charlotte. AND HELLO BRIAN!"`

### Requirement 7
If any entries in `name` are a string containing a comma, split it as its own input. For example, when `name` is `["Bob", "Charlie, Dianne"]`, then the method should return the string `"Hello, Bob, Charlie and Dianne."`

### CODE REVIEW

File: test_main.py [hunk 1]
```diff
@@ -22,3 +22,8 @@ def test_requirement_03(name):
 @pytest.mark.parametrize('name', [['Bruce Wayne', 'Selina Kyle'], ['Clark', 'Lois']])
 def test_requirement_04(name):
     assert greet(name) == f'Hello, {name[0]} and {name[1]}.'
+
+
+@pytest.mark.parametrize('name', [['Jean Grey', 'Kurt', 'Hank', 'Charles'], ['Scott Summers', 'Wade Wilson', 'Logan']])
+def test_requirement_05(name):
+    assert greet(name) == f'Hello, {", ".join(name[:-1])} and {name[-1]}.'
```
Comment (artsiom-streetshares) [inline test_main.py:29]: the problem here that you aren't testing that `{", ".join(name[:-1])} and {name[-1]}` does what you expect. I would remove parametrize in favor of having cleaner test.

File: main.py [hunk 2]
```diff
@@ -1,13 +1,28 @@
...
 def greet(name: [str, list, None] = None) -> str:
     """ Compose greeting string """
+
+    shouting = ''
+
     if isinstance(name, list):
-        name = ' and '.join(name)
+        names = []
+
+        for n in name:
```
Comment (artsiom-streetshares) [inline main.py:10]: avoid using one letter variables. it's fine to use in list comprehensions though, when scope is tiny

File: test_main.py (hunk 1 above)
Comment (artsiom-streetshares) [reply test_main.py:29]: or rather provide expected as you did below

Comment (s10-lee) [reply test_main.py]: ok. I got it.

**Review by artsiom-streetshares — APPROVED** [review]

**Pull request merged**